src/main/java/uk/gov/companieshouse/company_appointments/CompanyAppointmentService.java

```diff
@@ -55,13 +55,21 @@ public AllCompanyAppointmentsView fetchAppointmentsForCompany(String companyNumb
         }
 
         List<CompanyAppointmentView> companyAppointmentViews = allAppointmentData.stream().map(companyAppointmentMapper::map).collect(Collectors.toList());
-        int activeCount = (int) companyAppointmentViews.stream().filter(officer -> officer.getResignedOn() == null).count();
+
+        int activeCount = 0;
+        int inactiveCount = 0;
+
+        if (allAppointmentData.get(0).getCompany_status().equals("active")){
+            activeCount = (int) companyAppointmentViews.stream().filter(officer -> officer.getResignedOn() == null).count();
+        } else {
+            inactiveCount = (int) companyAppointmentViews.stream().filter(officer -> officer.getResignedOn() == null).count();
```

Comment:
In order to avoid the repeated stream line you could set a count value above the if to the stream output and then in the if block just set active or inactive count to the value of count?